setup.py

```diff
@@ -5,55 +5,32 @@
 Email: yli056@e.ntu.edu.sg
 Date: 9/19/2020
 
-# TODO: use pre-install to pip install requests first, may try this: https://stackoverflow.com/a/36902139
 """
-import platform
-import subprocess
+import os
 import sys
 import tarfile
+import warnings
 from distutils.core import setup
 from pathlib import Path
 
+import distro
 import requests
 from setuptools import find_packages
 
 ################################################################################
-# Check Platform and Python Version
-################################################################################
-
-if sys.version_info < (3,):
-    print('Python 2 has reached end-of-life and is no longer supported by PyTorch.')
-    sys.exit(-1)
-if sys.platform != 'linux':
-    TRITON_CLIENT_INSTALL = False
-else:
-    TRITON_CLIENT_INSTALL = True
-
-python_min_version = (3, 7, 0)
```

Comment:
Does this pip installation check the Python version? It is not able to run on versions below 3.7

Reply:
It's weird that pip installation won't stop by specifying `python_requires`, I'll recover this part of code